### PR TITLE
Fix handling group update

### LINKED ISSUE #3415: Remove contact from group
### Steps to reproduce

1. Go to Contact App
2. Add new contact
3. Scroll to add to group and add contact to multiple group
4. Now refresh page and edit same contact
5. try to remove contact from one group
6. It remove contact from all group

### Expected behavior

need to be removed from one selected group only

### Actual behavior

removes contact from all group

### Server operating system

_No response_

### Web server

Nginx

### Database engine version

None

### PHP engine version

None

### Nextcloud version

25.0.6

### Updated from an older installed version or fresh install

None

### Signing status

_No response_

### List of activated apps

_No response_

### Nextcloud configuration

_No response_

### Browser

_No response_

### Client operating system

_No response_

### CardDAV-clients

_No response_

### Web server error log

_No response_

### Nextcloud log

_No response_

### Browser log

_No response_

### Browser log

_No response_

### LINKED ISSUE #3446: No group contact not saved
### Describe the bug

When in the "no group" list in Contacts, if there is one last contact in the list and it is being edited, assigning it a group will immediately make it disappear from the list and also from the edition screen. The "no group" list is now blank which seems logical, but the contact actually still has no group. Modification has not been saved.

### Steps to reproduce

For this issue to be reproduced, there need to be only one contact with no group assigned.

1. Go to 'Contacts'
2. Click on 'No group' (don't know the exact name in english)
3. Click on the contact in the list
4. Update the contact and assign a group
5. The edition of the contact is automatically closed and "There is currently no contact" is displayed
6. Go back to the contact which was being edited. Modifications are lost and there is still no group

### Expected behavior

When assigning a group in this particular case, the edition should still be open until the user chooses to close or save it.

### Actual behavior

The edition closes automatically and nothing is saved.

### Contact version

5.3.0

### Operating system

Dbian GNU/Linux 11

### PHP engine version

Other

### Web server

Apache (supported)

### Database

MariaDB

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by hamza221**
fix  #3446 

The state was changing on local value update.
open for alternative, cleaner?  solution ideas 🧐

### CODE REVIEW

File: src/components/Properties/PropertyGroups.vue [hunk 1]
```diff
@@ -39,15 +39,15 @@
 					:no-wrap="true"
 					:placeholder="t('contacts', 'Add contact in group')"
 					:multiple="true"
-					:taggable="true"
 					:close-on-select="false"
 					:disabled="isReadOnly"
-					:tag-width="60"
+					:clearable="true"
+					:deselectFromDropdown="true"
+					:taggable:="true"
```
Comment (st3iny) [inline src/components/Properties/PropertyGroups.vue:46]:
```suggestion
					:taggable="true"
```

File: src/components/Properties/PropertyGroups.vue [hunk 2]
```diff
@@ -145,54 +145,17 @@ export default {
...
-				contact: this.contact,
-				groupName,
-			})
-			this.updateValue()
-		},
 
 		/**
 		 * Dispatch contact removal from group
 		 *
 		 * @param {string} groupName the group name
```
Comment (st3iny) [inline src/components/Properties/PropertyGroups.vue:153]:
```suggestion
		 * Send update event to parent
```

File: src/components/Properties/PropertyGroups.vue [hunk 3]
```diff
@@ -145,54 +145,17 @@ export default {
...
-			this.$store.dispatch('removeContactToGroup', {
-				contact: this.contact,
-				groupName,
-			})
-			const group = this.$store.getters.getGroups.find(search => search.name === groupName)
-			if (group.contacts.length === 0) {
-				this.$emit('update:value', [])
-			}
+		updateValue(groupNames) {
+			this.$emit('update',this.localValue)
```
Comment (st3iny) [inline src/components/Properties/PropertyGroups.vue:156]:
```suggestion
		updateValue() {
			this.$emit('update:value', this.localValue)
```

File: src/store/groups.js [hunk 4]
```diff
@@ -126,6 +126,11 @@ const getters = {
 
 const actions = {
 
+	updateContactGroup(context, { groupNames, contact }){
+		context.commit('removeContactFromGroups', contact)
+		context.commit('addContactToGroups', { groupNames, contact })
+
```
Comment (st3iny) [inline src/store/groups.js:132]:
```suggestion
```

File: src/store/groups.js [hunk 5]
```diff
@@ -126,6 +126,11 @@ const getters = {
 
 const actions = {
 
+	updateContactGroup(context, { groupNames, contact }){
```
Comment (st3iny) [inline src/store/groups.js:129]:
```suggestion
	updateContactGroups(context, { groupNames, contact }) {
```

File: src/components/ContactDetails.vue [hunk 6]
```diff
@@ -257,10 +257,11 @@
 
 				<!-- Groups always visible -->
 				<PropertyGroups :prop-model="groupsModel"
-					:value.sync="localContact.groups"
+					:value="localContact.groups"
 					:contact="contact"
 					:is-read-only="isReadOnly"
-					class="property--groups property--last" />
+					class="property--groups property--last" 
+					@update="updateGroups"/>
```
Comment (st3iny) [inline src/components/ContactDetails.vue:264]:
```suggestion
					@update:value="updateGroups"/>
```

File: src/components/ContactDetails.vue [hunk 7]
```diff
@@ -857,6 +862,11 @@ export default {
 		 */
 		async onSave() {
 			try {
+				this.localContact.groups = [...this.newGroupsValue]
+				await this.$store.dispatch('updateContactGroup', {
```
Comment (st3iny) [inline src/components/ContactDetails.vue:866]:
```suggestion
				await this.$store.dispatch('updateContactGroups', {
```

**Review by st3iny — COMMENTED** [review]
Some minor feedback. There are more cosmetic issues that should be automatically fixed by the linter (`npm run eslint:fix`).

I'm currently testing the PR.

File: src/components/Properties/PropertyGroups.vue [hunk 8]
```diff
@@ -39,15 +39,15 @@
...
 					:placeholder="t('contacts', 'Add contact in group')"
 					:multiple="true"
-					:taggable="true"
 					:close-on-select="false"
 					:disabled="isReadOnly"
-					:tag-width="60"
+					:clearable="true"
+					:deselectFromDropdown="true"
+					:taggable:="true"
+					:limit="3"
```
Comment (st3iny) [inline src/components/Properties/PropertyGroups.vue:47]: Why limit to 3? This means that a user is not able to see all groups in the dropdown.

**Review by st3iny — COMMENTED** [review]
I tested the repro from #3415 and the PR fixes the issue.

**Comment by st3iny** [comment]
Does it fix #3415?

**Comment by hamza221** [comment]
> Does it fix #3415?

I was not able to reproduce the bug, So I'm not sure

File: src/components/Properties/PropertyGroups.vue (hunk 8 above)
Comment (st3iny) [reply src/components/Properties/PropertyGroups.vue:47]: Which comment? IMO, the limit should be removed.

**Review by st3iny — CHANGES_REQUESTED** [review]
Tested and works. Please address https://github.com/nextcloud/contacts/pull/3452#discussion_r1227656573.

**Review by JohannesGGE — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Fixup check fails because of https://github.com/nextcloud/contacts/issues/3476

**Pull request merged**